coral/src/app/features/components/filters/SearchFilter.tsx

```diff
@@ -1,23 +1,38 @@
-import { SearchInput } from "@aivenio/aquarium";
+import { Box, Label, SearchInput, Typography } from "@aivenio/aquarium";
 import debounce from "lodash/debounce";
 import type { ChangeEvent } from "react";
 import { useFiltersContext } from "src/app/features/components/filters/useFiltersContext";
 
 type SearchFilterProps = {
+  label: string;
   placeholder: string;
   description: string;
+  ariaDescription: string;
 };
 
-function SearchFilter({ placeholder, description }: SearchFilterProps) {
+function SearchFilter({
+  label,
+  placeholder,
+  description,
+  ariaDescription,
+}: SearchFilterProps) {
   const { search, setFilterValue } = useFiltersContext();
   return (
-    <>
+    <Box maxWidth={"md"}>
+      <Label id={"search-field-label"}>
+        <Typography.SmallStrong color={"grey-60"}>
+          {label}
+        </Typography.SmallStrong>
+      </Label>
+
       <SearchInput
         type={"search"}
         aria-label={placeholder}
+        aria-labelledby={"search-field-label"}
         aria-describedby={"search-field-description"}
         role="search"
         placeholder={placeholder}
+        aria-description={ariaDescription}
```

Comment:
We can remove this now. Just in case it's already supported _somewhere_ - I'm not sure if it then would replace the aria-describedby (or read it double)

Reply:
[f0b9edb](https://github.com/Aiven-Open/klaw/pull/1898/commits/f0b9edb38167d70087488ceb934298e2748d6ff8)